db/table_cache.cc

```diff
@@ -203,7 +203,7 @@ InternalIterator* TableCache::NewIterator(
     if (create_new_table_reader) {
       unique_ptr<TableReader> table_reader_unique_ptr;
       s = GetTableReader(
-          env_options, icomparator, fd, true /* sequential_mode */, readahead,
+          env_options, *icomparator, fd, true /* sequential_mode */, readahead,
```

Comment:
Why change it from reference to pointer everywhere? Why exclude GetTableReader() from this change?